app/controllers/members.js

```diff
@@ -2,23 +2,88 @@ import Controller from '@ember/controller';
 import ghostPaths from 'ghost-admin/utils/ghost-paths';
 import moment from 'moment';
 import {computed} from '@ember/object';
+import {get} from '@ember/object';
+import {pluralize} from 'ember-inflector';
 import {inject as service} from '@ember/service';
 import {task} from 'ember-concurrency';
 
 /* eslint-disable ghost/ember/alias-model-in-controller */
 export default Controller.extend({
     store: service(),
 
+    queryParams: ['label'],
+
+    label: null,
     members: null,
     searchText: '',
+    modalLabel: null,
+    showLabelModal: false,
+
+    _hasLoadedLabels: false,
+    _availableLabels: null,
 
     init() {
         this._super(...arguments);
         this.set('members', this.store.peekAll('member'));
+        this._availableLabels = this.store.peekAll('label');
     },
 
-    filteredMembers: computed('members.@each.{name,email}', 'searchText', function () {
-        let {members, searchText} = this;
+    listHeader: computed('selectedLabel', 'searchText', function () {
+        let {searchText, selectedLabel, filteredMembers} = this;
+        if (searchText) {
+            return 'Search result';
+        }
+        if (this.fetchMembers.lastSuccessful) {
+            let count = pluralize(filteredMembers.length, 'member');
+            if (selectedLabel && selectedLabel.slug) {
+                if (filteredMembers.length > 1) {
+                    return `${count} match current filter`;
+                } else {
+                    return `${count} matches current filter`;
+                }
+            }
+            return count;
+        }
+        return 'Loading...';
+    }),
+
+    showingAll: computed('label', 'searchText', function () {
+        let {searchText, label} = this;
+
+        return !searchText && !label;
+    }),
+
+    availableLabels: computed('_availableLabels.@each.{isNew}', function () {
+        let labels = this._availableLabels
+            .filter(label => !label.get('isNew'))
+            .filter(label => label.get('id') !== null)
+            .sort((labelA, labelB) => labelA.name.localeCompare(labelB.name, undefined, {ignorePunctuation: true}));
+        let options = labels.toArray();
+
+        options.unshiftObject({name: 'All labels', slug: null});
+
+        return options;
+    }),
+
+    selectedLabel: computed('label', 'availableLabels', function () {
+        let label = this.get('label');
+        let labels = this.get('availableLabels');
+
+        return labels.findBy('slug', label);
+    }),
+
+    labelModalData: computed('modalLabel', 'availableLabels', function () {
+        let label = this.get('modalLabel');
+        let labels = this.get('availableLabels');
+
+        return {
+            label,
+            labels
+        };
+    }),
```

Comment:
@kevinansfield So I am not sure if this was the best pattern to pick here, but this basically just tries to have the modal data we use for label add/edit modal, and this seemed like the best way to have it and pass it around. I was also confused if going custom views route with a service for labels is a good approach, and felt like it wasn't. Lmk what you think about it 😬